src/hyperdeck/response-handler.js

```diff
@@ -12,73 +12,78 @@ var ASYNC_RESPONSE_REGEX = /^(5[0-9]{2} )/;
 function ResponseHandler(clientSocket) {
   var destroyed = false;
   var notifier = new events.EventEmitter();
-  var buffer = "";
-  var nextResponseIsMultiline = false;
-  
-  function isBufferComplete() {
-    var lines = buffer.split("\r\n");
-    // there will always be an empty element at the end as every line will always end in "\r\n"
-    // so remove it.
-    // i.e. "1\r\n2\r\n" => ["1", "2", ""]
-    lines.pop();
-    if (lines.length === 1) {
-      var asyncResponse = ASYNC_RESPONSE_REGEX.exec(lines[0]);
-      if (!asyncResponse) {
-        if (nextResponseIsMultiline) {
-          logger.debug('Assuming buffer is not complete because nextResponseIsMultiline flag is set.');
-          nextResponseIsMultiline = false;
-          return false;
-        }
-      }
-      // is it a single line response?
-      return lines[0].indexOf(":") !== lines[0].length-1;
-    } 
-    // multi line response so waiting for a blank line to signify end
-    return lines[lines.length-1] === "";
+  var buffer = [];
+
+  function isRespComplete() {
+    let complete = false;
+
+    if (buffer.length === 1) {
+      // a single line response
+      // single line response first line should never ends with ":" and
+      // 1XX and 200 is always single line response
+      complete = buffer[0].search(/^1\d\d\s|200\sok$/) === 0 && buffer[0].endsWith(':') === false;
```

Comment:
We could do this in one regex instead (which we should put a the top of the file so it is cached).

```javascript
var SINGLE_LINE_REGEX = /^(?:1\d{2}|200) .*?[^\:]$/;
```

and do
```javascript
if (SINGLE_LINE_REGEX.test(buffer[0])) {
  return true;
}
```

Reply:
SINGLE_LINE_REGEX  is better but it should match [200\sok] not just [200].
reason for that if you have more than 200 files then [disk list] command response will contain 
200:[File Name] ....
so it will fail in that case.

Reply:
I think we are ok though because a filename would never be in the first line of the buffer right? The only valid time when we could get a filename would be when we are in a multiline response and waiting for an empty line?

Reply:
still [200\sok] is the safer way.

Reply:
I think it's actually safer the other way because that would already handle

200 something else

The same way we handle 

500 anything

Reply:
there will never be [ 200 something else ].
[ 200\sok ] is only one option for 200.

Reply:
if you open 200 then i cant guarantee about current logic, i need to rethink the whole solution to make sure it will work.

Reply:
Say
200 something
did happen though 

The outcome is either that we treat is as a single line, or expect multiple.

Don't you think it makes more sense that we guess its a single line? If we guess wrongly that it's multiple then we will loose some of the next commands. If we guess wrongly that it is single then we will probably just miss some data but we won't miss any future commands. I think there will be some errors but the end result would be future commands would not get missed.

Reply:
please read the documentation once more, they explicitly explained that 200\sok is a special response. there is no option for 200 something.

Reply:
The documentation is pretty vague in places though. They say
> ![image](https://user-images.githubusercontent.com/3259993/33171854-24b9c78e-d04e-11e7-8002-2bbf67b98cba.png)

so it doesn't actually mention that it will always be `ok`. Also they say 

> ![image](https://user-images.githubusercontent.com/3259993/33172005-b3c6e470-d04e-11e7-92f0-4496e3ec01f4.png)

which is wrong for the format command which doesn't have a response in the params format

I think it is better to just be as specific as needed in order for everything to work. Checking that the text is `ok` doesn't actually achieve anything (I don't think anything breaks?), and the downside is if in an update they change this wording for some reason it will break things.

After thinking about it more I think it would be even better to just do
```javascript
var SINGLE_LINE_REGEX = /^(?:1\d{2}|200) /;
```
given they are saying 200 is single line.